kratos/processes/calculate_embedded_nodal_variable_from_skin_process.h

```diff
@@ -723,27 +735,38 @@ class CalculateEmbeddedNodalVariableFromSkinProcess : public Process
 
     void AddEdgeNodes(
         const Geometry<Node<3>> &rEdgeGeometry,
-        ModelPart &rModelPart) const
+        ModelPart &rModelPart,
+        std::unordered_map<unsigned int, Node<3>::Pointer>& rNodesMap
+        ) const
     {
+        const auto& pvar_list = rModelPart.pGetNodalSolutionStepVariablesList();
+        unsigned int buffer_size = rModelPart.GetBufferSize();
+        
         // Loop the edge nodes
         for (std::size_t i = 0; i < 2; ++i) {
             auto p_i_node = rEdgeGeometry(i);
             // Check if the node has been already added
             if (!p_i_node->Is(VISITED)) {
                 p_i_node->Set(VISITED, true);
-                rModelPart.CreateNewNode(p_i_node->Id(), *p_i_node);
+                auto p_node_copy = Kratos::make_intrusive< Node<3> >( p_i_node->Id(), 
+                                                                      p_i_node->Coordinates()
+                                                                      );
```

Comment:
```suggestion
                auto p_node_copy = Kratos::make_intrusive< Node<3> >(
                    p_i_node->Id(), 
                    p_i_node->Coordinates());
```
(minor) I think it's easier to read in this way.

Reply:
I prefer everything in a line...